src/vision/status_quo.md

```diff
@@ -0,0 +1,79 @@
+
+Eventually, skimming on stack overflow and elsewhere, he comes across a link to the ["Asynchronous Programming in Rust" book]. In the book, he learns about Rust's async/await syntax. Using the book he [creates a "Hello, World" example] -- that's fun! He's feeling confident.
+
+Unfortunately, the book goes straight from there to low-level details on how futures are implemented, and that's confusing. The next few sections also seem to be kind of detailed. He just wants to make some working code! He skips ahead a bit.
+
+He gets to the [Async Ecosystem chapter] and he realizes that he has to pick a runtime. This is kind of stressful. What if he picks the wrong one? What if he starts building a project and then realizes that the runtime he chose isn't compatible with some library he wants to use?[^cite-FOMO] It's already been an hour or two, so he decides to just go to bed and stream [Bob Ross] on PBS instead. Rust will be there tomorrow.
+
+[Rust By Example]: https://doc.rust-lang.org/rust-by-example/
+[The Rust Book]: https://doc.rust-lang.org/book/
+["learning" section on rust-lang.org]: https://www.rust-lang.org/learn
+["Asynchronous Programming in Rust" book]: https://rust-lang.github.io/async-book/
+[creates a "Hello, World" example]: https://rust-lang.github.io/async-book/01_getting_started/04_async_await_primer.html
+[Async Ecosystem chapter]: https://rust-lang.github.io/async-book/08_ecosystem/00_chapter.html
+[Bob Ross]: https://www.pbs.org/show/best-joy-painting/
+
+## Bob has misleading intutions
+
+Bob eventually picks a runtime and starts working through the tutorial[^other_sections]. He extends his first async-await program to one that invokes a few helper routines and awaits them:
+
+```rust,edition2018
+# struct X { }
+# struct Y { }
+# struct Z { }
+# async fn first_thing() -> X { X { } }
+# async fn second_thing() -> Y { Y { } }
+# async fn third_thing() -> Z { Z { } }
+async fn do_the_thing() -> (X, Y, Z) {
+    let x = first_thing();
+    let y = second_thing();
+    let z = third_thing();
+    (x.await, y.await, z.await)
+}
+```
+
+When Bob runs the program, he is surprised to see that these three functions seem to run sequentially. That's strange, he thought that asynchronous functions were supposed to run...asynchronously? Bob is frustrated.
+
+Reading into the book a bit more, Bob learns that Rust's asynchronous functions are *lazy*, and they don't start running until they are awaited. He reads about the various options (`FuturesUnordered`, etc) and ultimately opts to add in some calls to `spawn` from his favorite runtime:
```

Comment:
I'm not against recommending spawn per-se, but again I'm pretty convinced the idiomatic way today is to use join (`join3!` in this case).